server/chainquery/queries/claimQueries.js

```diff
@@ -0,0 +1,184 @@
+    })
+    .then(channelClaimsArray => {
+      if(channelClaimsArray.length === 0) {
+        return null;
+      }
+
+      return channelClaimsArray;
+    })
+  },
+
+  getClaimIdByLongChannelId: async (channelClaimId, claimName) => {
+    logger.debug(`finding claim id for claim ${claimName} from channel ${channelClaimId}`);
+    return await table.findAll({
+      where: { name: claimName, publisher_id: channelClaimId },
+      order: [['id', 'ASC']],
+    })
+    .then(result => {
+      switch (result.length) {
+        case 0:
+          return null;
+        case 1:
+          return result[0].claim_id;
+        default:
+          // Does this actually happen??? (from converted code)
+          logger.warn(`${result.length} records found for "${claimName}" in channel "${channelClaimId}"`);
+          return result[0].claim_id;
+      }
+    });
+  },
+
+  validateLongClaimId: async (name, claimId) => {
+    return await table.findOne({
+      where: {
+        name,
+        claim_id: claimId,
+      },
+    }).then(result => {
+      if (!result) {
+        return false;
+      }
+      return claimId;
+    });
+  },
+
+  getLongClaimIdFromShortClaimId: async (name, shortId) => {
+    return await table.findAll({
+      where: {
+        name,
+        claim_id: {
+          [sequelize.Op.like]: `${shortId}%`,
+        }},
+      order: [['height', 'ASC']],
+    })
+    .then(result => {
+      if(result.length === 0) {
+        return null;
+      }
+
+      return result[0].claim_id;
+    });
+  },
+
+  getTopFreeClaimIdByClaimName: async (name) => {
+    return await table.findAll({
+      // TODO: Limit 1
+      where: { name },
+      order: [['effective_amount', 'DESC'], ['height', 'ASC']],
+    }).then(result => {
+      if(result.length === 0) {
+        return null;
+      }
+      return result[0].claim_id;
+    })
+  },
+
+  getLongClaimId: async (claimName, claimId) => {
+    // TODO: Add failure case
+    logger.debug(`getLongClaimId(${claimName}, ${claimId})`);
+    if (isLongClaimId(claimId)) {
+      return table.queries.validateLongClaimId(claimName, claimId);
+    } else if (isShortClaimId(claimId)) {
+      return table.queries.getLongClaimIdFromShortClaimId(claimName, claimId);
+    } else {
+      return table.queries.getTopFreeClaimIdByClaimName(claimName);
+    }
+  },
+
+  resolveClaim: async (name, claimId) => {
+    logger.debug(`Claim.resolveClaim: ${name} ${claimId}`);
+    return table.findAll({
+      where: { name, claim_id: claimId },
+    }).then(claimArray => {
+      if(claimArray.length === 0) {
+        return null;
+      } else if(claimArray.length !== 1) {
+        logger.warn(`more than one record matches ${name}#${claimId} in db.Claim`);
+      }
+
+      return claimArray[0];
+    });
+  },
+
+  getOutpoint: async (name, claimId) => {
+    logger.debug(`finding outpoint for ${name}#${claimId}`);
+
+    return await table.findAll({
+      where     : { name, claim_id: claimId },
```

Comment:
I think I get it. You are not specifying name because they are identical. However, the claim table is unique on `claim_id`. So you do not need to ever you both. There is no benefit that I can think of. If there is let me know.